### PR TITLE
ENT-7620: Added the ability to blacklist job-skill relationship.

### PR DESCRIPTION
**Pull request by saleem-latif**
__JIRA Ticket:__ [ENT-7620](https://2u-internal.atlassian.net/browse/ENT-7620)

__Drscription:__
We want the ability to blacklist/whitelist jobs skills, We can achieve that by introducing a new field is_blacklisted in [BaseJobSkill](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/models.py#L596C7-L596C19) model.
**Acceptance Criteria:**

1. is_blacklisted is added to [BaseJobSkill](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/models.py#L596C7-L596C19) model with a default of False.
2. QuerySets such is whitelisted_objects and blacklisted_objectsare added for [BaseJobSkill](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/models.py#L596C7-L596C19) model.
3. Utility methods such as get_whitelisted_job_skills, blacklist_job_skill are added to [BaseJobSkill](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/models.py#L596C7-L596C19) model

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/openedx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -13,6 +13,10 @@ Change Log
 
 Unreleased
 
+[1.45.0] - 2023-09-20
```
Comment (mahamakifdar19) [inline CHANGELOG.rst:16]: Shouldn't we change the date for this version?

**Review by mahamakifdar19 — APPROVED** [review]
Aside from modifying version date, LGMT!

File: taxonomy/models.py [hunk 2]
```diff
@@ -640,6 +641,24 @@ class Meta:
...
+
+    @classmethod
+    def get_blacklist_job_skill(cls):
+        """
+        Get a QuerySet of whitelisted job skills.
+
+        White listed job skills are job skills with `is_blacklisted=False`.
+        """
+        return cls.objects.filter(is_blacklisted=True)
+
```
Comment (muhammad-ammar) [inline taxonomy/models.py:661]: @saleem-latif Can we use a custom model manager in place of the above methods?

**Review by jajjibhai008 — APPROVED** [review]
LGTM 👍

File: taxonomy/models.py (hunk 2 above)
Comment (saleem-latif) [reply taxonomy/models.py:661]: I looked into that approach but it creates more problems. e.g. methods like `create_or_update`, `get_or_create` will not work correctly if we override the default model manager in our use case.

**Review by muhammad-ammar — APPROVED** [review]

**Pull request merged**